packages/appsync-modelgen-plugin/src/__tests__/visitors/__snapshots__/appsync-swift-visitor.test.ts.snap

```diff
@@ -0,0 +1,264 @@
+      self.id = id
+      self.name = name
+      self.list = list
+      self.requiredList = requiredList
+      self.requiredListOfRequired = requiredListOfRequired
+      self.listOfRequired = listOfRequired
+      self.requiredListOfRequiredDates = requiredListOfRequiredDates
+      self.listOfRequiredFloats = listOfRequiredFloats
+      self.requiredListOfCustomTypes = requiredListOfCustomTypes
+      self.createdAt = createdAt
+      self.updatedAt = updatedAt
+  }
+}"
+`;
+
+exports[`AppSyncSwiftVisitor Should handle nullability of lists appropriately 2`] = `
+"// swiftlint:disable all
+import Amplify
+import Foundation
+
+extension ListContainer {
+  // MARK: - CodingKeys 
+   public enum CodingKeys: String, ModelKey {
+    case id
+    case name
+    case list
+    case requiredList
+    case requiredListOfRequired
+    case listOfRequired
+    case requiredListOfRequiredDates
+    case listOfRequiredFloats
+    case requiredListOfCustomTypes
+    case createdAt
+    case updatedAt
+  }
+  
+  public static let keys = CodingKeys.self
+  //  MARK: - ModelSchema 
+  
+  public static let schema = defineSchema { model in
+    let listContainer = ListContainer.keys
+    
+    model.pluralName = \\"ListContainers\\"
+    
+    model.fields(
+      .id(),
+      .field(listContainer.name, is: .optional, ofType: .string),
+      .field(listContainer.list, is: .optional, ofType: .embeddedCollection(of: Int.self)),
+      .field(listContainer.requiredList, is: .required, ofType: .embeddedCollection(of: String.self)),
+      .field(listContainer.requiredListOfRequired, is: .required, ofType: .embeddedCollection(of: StatusEnum.self)),
+      .field(listContainer.listOfRequired, is: .optional, ofType: .embeddedCollection(of: Bool.self)),
+      .field(listContainer.requiredListOfRequiredDates, is: .required, ofType: .embeddedCollection(of: Temporal.Date.self)),
+      .field(listContainer.listOfRequiredFloats, is: .optional, ofType: .embeddedCollection(of: Double.self)),
+      .field(listContainer.requiredListOfCustomTypes, is: .required, ofType: .embeddedCollection(of: CustomType.self)),
+      .field(listContainer.createdAt, is: .optional, isReadOnly: true, ofType: .dateTime),
+      .field(listContainer.updatedAt, is: .optional, isReadOnly: true, ofType: .dateTime)
+    )
+    }
+}"
+`;
+
+exports[`AppSyncSwiftVisitor Should handle nullability of lists appropriately 3`] = `
+"// swiftlint:disable all
+import Amplify
+import Foundation
+
+public struct CustomType: Embeddable {
+  var name: String?
+  var list: [Int?]?
+  var requiredList: [String?]
+  var requiredListOfRequired: [StatusEnum]
+  var listOfRequired: [Bool]?
+}"
+`;
+
+exports[`AppSyncSwiftVisitor Should render lists with HAS_MANY connection according to the spec 1`] = `
+"// swiftlint:disable all
+import Amplify
+import Foundation
+
+public struct Todo: Model {
+  public let id: String
+  public var title: String
+  public var tasks: List<task?>
```

Comment:
pay attention to this type. Since the schema has `tasks: [task]! @connection(name: "TodoTasks")`, the list is required here.